### PR TITLE
Integration Blog Article

### PR DESCRIPTION
**Pull request by carnellj**

### CODE REVIEW

File: source/2020-10-19-integrating-with-the-cloud/index.html.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+tags: Genesys Cloud, Developer Engagement
+date: 2020-10-19
+author: john.carnell@genesys.com
+---
+
+Integration. It's not the most fun part of software development, but it is one of the more critical things you need to think about as you are looking at purchasing a cloud-based product. Cloud products are great in the sense that you don't have to manage the infrastructure yourself, you get new features quickly, and you are able to convert a fixed capital cost over to a variable expense (for the MBAs in the audience). However, the success or failure of your cloud acquisition is going to be tied to how successful you are in integrating your new cloud platform into your overall business processes/workflows and the IT ecosystem of your organization. You need to understand that there are different styles of integrations and every cloud provider will give you different technologies to use with these styles. 
+
+This article is going to walk through what are the different styles of cloud integration and talk specifically about which Genesys Cloud technologies are available for each integration style.  
+
+:::Some thoughts on integration styles
```
Comment (PrinceNebulon) [inline source/2020-10-19-integrating-with-the-cloud/index.html.md:12]: Need to use the advanced syntax to get a title: https://apicentral.inindca.com/meta/authoring/alerts

File: source/2020-10-19-integrating-with-the-cloud/index.html.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+There are a lot of different mechanisms for pulling data out of a cloud provider. However, when we talk specifically about how we do it in Genesys Cloud, the topic of data integration centers around the Genesys Cloud [Analytics API](/api/rest/v2/analytics/overview.html) and event-based [Notifications](/api/rest/v2/notifications/notification_service.html).  
+
+### Polling for Data with the Analytics API
+
+The Analytics API is not a monolithic API, but rather encompasses three levels of data detail:
+
+1. [Observations](api/rest/v2/analytics/overview.html#instantaneous_observations_metrics).  The Observation APIs return information about the current state of objects within Genesys Cloud. Observation queries are meant to be a point in time snapshot of what is going on in your contact center. They offer limited filtering and no date-based queries.
```
Comment (PrinceNebulon) [inline source/2020-10-19-integrating-with-the-cloud/index.html.md:82]: link needs a leading slash

File: source/2020-10-19-integrating-with-the-cloud/index.html.md (hunk 1 above)
Comment (carnellj) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:12]: Shoot.  Let me push as I already fixed this

File: source/2020-10-19-integrating-with-the-cloud/index.html.md (hunk 2 above)
Comment (carnellj) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:82]: Cool.  Good catch and fixed.

File: source/2020-10-19-integrating-with-the-cloud/index.html.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+So how do you deal with situations where you have low-data latency data requirements? Polling for this data using the Analytics API introduces timeout problems and using the `Jobs` endpoints means you can not retrieve the most current data. Genesys Cloud offers an event-based message stream that will publish conversation and user detail messages at the time the event occurs. There are two mechanisms for consuming these messages streams:
+
+1. The WebSocket-Based Notification Services
+2. AWS EventBridge
+
+The WebSocket-based [Notifications](/api/rest/v2/notifications/notification_service.html) service allows you to open a web socket to Genesys Cloud and subscribe to events as they occur. Events will be published to the WebSocket. This web socket does not guarantee message delivery and if the web socket is disconnected at the time message is published, it will be lost. This puts the onus on you as the developer to maintain socket state and to build data recovery code if the socket becomes unavailable. This is usually done by using the `Details` API to retrieve any data that was missed from when the web socket went down and when a new one was established.
```
Comment (PrinceNebulon) [inline source/2020-10-19-integrating-with-the-cloud/index.html.md:98]: > This is usually done by using the `Details` API to retrieve any data that was missed from when the web socket went down and when a new one was established.

I'm not sure to what this is referring. You can't go back and get missed WebSocket events; the only way I'm aware of is to use whatever APIs provide the same data as the notification. That association is now documented here: https://developer.mypurecloud.com/api/rest/v2/notifications/available_topics.html

File: source/2020-10-19-integrating-with-the-cloud/index.html.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## Presentation Integration Style
+
+The last integration style is presentation integration. This type of integration involves embedding user interface components from the cloud provider inside your own application or embedding your own application components inside the cloud provider's tool. Every cloud provider is going to offers a wide variety of options to do this. Genesys Cloud categorizes their presentation integration capabilities into two buckets:
+
+1. Agent based UI integrations
+2. End User based UI integrations
+
+For agent-based UI, Genesys cloud offers a number of different capabilities. For embedding custom applications within Genesys Cloud, you can leverage the [Client Apps](/api/client-apps/) framework. This framework allows you to embed an entire application as a sidebar or full frame application within the Genesys Cloud user interface. For embedding Genesys Cloud functionality inside your own applications, you can leverage the [Embeddable](/api/embeddable-framework/). In addition, if you want to embed Genesys Cloud phone controls via WebRTC into your own applications, you can leverage the [WebRTC SDK](/api/webrtcsdk/).
```
Comment (PrinceNebulon) [inline source/2020-10-19-integrating-with-the-cloud/index.html.md:111]: link should be named `[Embeddable Framework]`

File: source/2020-10-19-integrating-with-the-cloud/index.html.md (hunk 3 above)
Comment (carnellj) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:98]: What I am getting at here and I should take another cut is that during the time period the websocket is down, you can still use the Details API to retrieve any data conversation data from that time period and rebuild your view of the data.  

I will clean up this section to make that a little bit clearer

Comment (PrinceNebulon) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:98]: A little clarification wouldn't hurt; my comment was mainly a nitpick about saying "Details API". You're not going to find anything by that name here: https://developer.mypurecloud.com/api/rest/v2/. The concept of rebuilding missed data applies to any notification topics, not just conversation details (if that's what you're referring to), so the API resources you'll use to rebuild your data are different depending on which notification topics you're using.

File: source/2020-10-19-integrating-with-the-cloud/index.html.md (hunk 4 above)
Comment (carnellj) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:111]: Done

File: source/2020-10-19-integrating-with-the-cloud/index.html.md (hunk 3 above)
Comment (carnellj) [reply source/2020-10-19-integrating-with-the-cloud/index.html.md:98]: Done

**Review by PrinceNebulon — APPROVED** [review]

**Pull request merged**